session_schema.go

```diff
@@ -245,12 +245,17 @@ func (session *Session) Sync2(beans ...interface{}) error {
 		if err != nil {
 			return err
 		}
-		tbName := engine.TableName(bean)
-		tbNameWithSchema := engine.TableName(tbName, true)
+		var tbName string
+		if len(session.statement.AltTableName) > 0 {
+			tbName = session.statement.AltTableName
+		} else {
+			tbName = engine.TableName(bean)
+		}
+		tbNameWithSchema := engine.tbNameWithSchema(tbName)
 
 		var oriTable *core.Table
 		for _, tb := range tables {
-			if strings.EqualFold(tb.Name, tbName) {
+			if strings.EqualFold(engine.tbNameWithSchema(tb.Name), engine.tbNameWithSchema(tbName)) {
```

Comment:
Why repeat `engine.tbNameWithSchema(tbName)`? Isn't that just the same as `tbNameWithSchema` from above?

Reply:
`tb.Name` may have not schema but `tbName` may have or not.
`tbNameWithSchema` will check if tbname has schema, if not then add it.

Reply:
I meant this one:
```
254    ==>this tbNameWithSchema := engine.tbNameWithSchema(tbName)
```

vs. 

```
258    if strings.EqualFold(engine.tbNameWithSchema(tb.Name), ==>this engine.tbNameWithSchema(tbName)) {
```

Reply:
Just know this after I merged. :(